### PR TITLE
Decorator for Node GUIs and GUI deferred loading

### PR DESCRIPTION
**Pull request by HeftyCoder**
This PR introduces a new way to assign Node GUIs to Nodes. Specifically:

- It introduces a decorator called `node_gui` which accepts a Node as a parameter. Assigning a gui to a node happens in the `gui.py` file and not the `nodes.py`, allowing for IDE hinting and deprecating the need for the `import_guis` and `export_guis` functions.
- Introduces a `on_gui_load` function which can be applied to any function in the imported `nodes.py` file.  This function will then be called after package loading in `import_nodes_package`.
- ~~Adds a `api.py` file in the ryven folder. Might not be needed, but thought it was a good idea.~~ Bad idea, read the conversation for details.

The `built_in` package and the `pkg_test` package (which has `linalg` and `std` subpackages) have been changed to use this new utility. The older `linalg` and `std` packages haven't been altered and are working with the "deprecated" functions.

CC: these ideas were first discussed in #171 as a result of investigating sub-packages

### CODE REVIEW

**Comment by leon-thomm** [comment]
Please correct me if I'm missing something but I think your `api.py` compromises the whole point of separating nodes dependencies and UI dependencies in the first place, which is that Ryven has a headless mode in which it runs without any Qt dependencies (only ryvencore).

**Comment by leon-thomm** [comment]
Clarifying the previous comment: Ryven can run in environments where Qt is not even installed (because it can't without desktop environment). That's why `ryven.main` is separate from `ryven.gui`. [^1] `ryven.main.Ryven.run()` runs with the editor, `ryven.main.RyvenConsole.run()` runs without.

[^1]: `config.py` imports something from `gui` but it doesn't cause `Qt`  import

File: ryven-editor/ryven/main/packages/nodes_package.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from ryvencore import Node, Data
 
 from ryven.main.utils import read_project, ryven_dir_path, abs_path_from_package_dir
-
+from .gui_env import in_gui_mode, load_current_guis
```
Comment (leon-thomm) [inline ryven-editor/ryven/main/packages/nodes_package.py:16]: same here, you should not import anything from `ryven.gui` in `ryven.main` unless you know you're running in GUI mode

**Comment by leon-thomm** [comment]
I took the liberty to fix the imports and remove `api`. If I messed up your local branch I can rollback re-commit later. Looks good to me now, anything else from your side?

**Comment by HeftyCoder** [comment]
> I think your api.py compromises the whole point of separating nodes dependencies and UI dependencies in the first place

I was (and probably still am) confused about how python importing works. I thought that by calling `from ryven.api import in_gui_mode` only that function would be imported, ignoring the rest. Didn't account that `in_gui_mode` was also imported in `gui_env`. I thought one could bundle everything that way. My bad.

I made minor changes to the `consume_last_exported_package` function. Other than that, I'm good!

**Review by leon-thomm — APPROVED** [review]

**Comment by leon-thomm** [comment]
Importing a module just means running it. A module-level expression such as `def foo` makes the `foo` symbol available. A module-level expression such as `import ryven.gui_env` runs `ryven.gui_env` at the point where the import is stated. A `from` import is actually no different, `from ryven.api import in_gui_mode` will first run `ryven.api` and then make `in_gui_mode` available locally. That's why it matters how imports are placed.

**Pull request merged**